TeensyDmx.cpp

```diff
@@ -534,7 +532,9 @@ uint16_t TeensyDmx::rdmSetDeviceLabel(struct RDMDATA* rdm)
         return E120_NR_FORMAT_ERROR;
     } else {
         memcpy(m_deviceLabel, rdm->Data, rdm->DataLength);
-        m_deviceLabel[rdm->DataLength] = '\0';
+        if (rdm->DataLength < RDM_MAX_STRING_LENGTH) {  // Don't overflow our buffer
```

Comment:
If you're checking here, you should probably check before doing a memcpy to the rest of it... I'd prefer to have the null byte at the end and truncate the input.

Reply:
I think that's rather against the spirit of the standard:
"Unless otherwise specified the text string length shall be variable up to 32 characters."
"The Parameter Data Length shall accordingly be set to match the actual size of the text string
being sent. Parsing of the string shall be length terminated corresponding to the Parameter Data
Length field. Text fields shall terminate based on Parameter Data Length, however if a NULL is
encountered then that shall also act as a terminator for the text field."

Although there is also:
"Responders that have limited capabilities may truncate and store less than 32 characters if
necessary."

Essentially they aren't required to have a null and truncating isn't great either. So our choices are to either do this or expand our buffer to store 33 characters and then strip the null off where necessary.

Reply:
Also we already check in https://github.com/chrisstaite/TeensyDmx/pull/21/files/c42b28c60a1ddb06b6b674abb30093f4b05d0f65#diff-432f612a4ce7dde2210eb7fe06f72f41R530 that we've only got 32 characters or less, so we won't overflow there.

Reply:
Quite, so storing 33 would be best I think.

Reply:
It's more the function const char* TeensyDmx::getLabel() const that I'm worried about...

Reply:
I'd missed that one.

Reply:
Done